src/dist/manifestation/tests.rs

```diff
@@ -992,13 +1019,24 @@ fn unavailable_components_is_target() {
             )
             .unwrap_err();
             match err.downcast::<RustupError>() {
-                Ok(e @ RustupError::RequestedComponentsUnavailable { .. }) => {
-                    let err_str = e.to_string();
-                    assert!(err_str
-                        .contains("rustup target remove --toolchain nightly i686-apple-darwin"));
-                    assert!(err_str.contains(
-                        "rustup target remove --toolchain nightly i686-unknown-linux-gnu"
-                    ));
+                Ok(RustupError::RequestedComponentsUnavailable {
+                    components,
+                    manifest,
+                    toolchain,
+                }) => {
+                    assert_eq!(toolchain, "nightly");
+                    let descriptions = components
+                        .iter()
+                        .map(|c| c.description(&manifest))
+                        .collect::<Vec<_>>();
+                    assert_eq!(
+                        descriptions,
+                        [
+                            "'rust-std' for target 'x86_64-apple-darwin'",
+                            "'rust-std' for target 'i686-apple-darwin'",
+                            "'rust-std' for target 'i686-unknown-linux-gnu'"
+                        ]
+                    );
```

Comment:
This actually exposes an error in the original test case, which only tested the INCLUSION rather than the whole collection.